src/library/Pagination/Pages.js

```diff
@@ -0,0 +1,156 @@
+  color_disabled: theme.color_theme
+}));
+
+const Buttons = ({
+  currentPage,
+  handleClick,
+  size,
+  totalPages,
+  visibleRange
+}: Props) => {
+  return [...Array(totalPages)]
+    .map((_, i) => i)
+    .map((_, index) => {
+      const page = index + 1;
+      const evenOffset = visibleRange % 2 === 0 ? 1 : 0;
+      let pagesBuffer;
+      if (
+        currentPage <= visibleRange ||
+        (visibleRange === 1 && currentPage === 2)
+      ) {
+        pagesBuffer = visibleRange - currentPage + 3 - evenOffset;
+      } else if (
+        currentPage > totalPages - visibleRange ||
+        (visibleRange === 1 && currentPage === totalPages - 1)
+      ) {
+        pagesBuffer =
+          visibleRange - (totalPages - currentPage) + 2 - evenOffset;
+      } else {
+        pagesBuffer = Math.ceil(visibleRange / 2);
+      }
+
+      const firstPageInRange =
+        page === currentPage - pagesBuffer && !firstPage(page);
+      const lastPageInRange =
+        page === currentPage + pagesBuffer && !lastPage(page, totalPages);
+      const isPageOutOfRange =
+        page < currentPage - pagesBuffer || page > currentPage + pagesBuffer;
+
+      let pageView = null;
+      if (firstPageInRange || lastPageInRange) {
+        pageView = (
+          <EllipsisButton
+            disabled
+            element="span"
+            key={page}
+            minimal
+            size="medium">
+            …
+          </EllipsisButton>
+        );
+      } else if (
+        !isPageOutOfRange ||
+        firstPage(page) ||
+        lastPage(page, totalPages)
+      ) {
+        pageView = (
+          <Button
+            minimal
+            primary={currentPage === page}
+            key={page}
+            onClick={handleClick.bind(null, page)}
+            size={size}>
+            {page}
+          </Button>
+        );
+      }
+      return pageView;
+    })
+    .filter((page) => !!page);
+};
+
+const GenericIncrementButton = ({
+  currentPage,
+  direction,
+  handleIncrement,
+  messages,
+  size,
+  totalPages
+}: Props & { direction: string }) => {
+  const incrementForward = direction === 'next' ? true : false;
+  const incrementIcon =
+    direction === 'next' ? <IconChevronRight /> : <IconChevronLeft />;
+  const disabled =
+    direction === 'next'
+      ? lastPage(currentPage, totalPages)
+      : firstPage(currentPage);
+
+  const iconPosition = direction === 'next' ? 'iconEnd' : 'iconStart';
+
+  const buttonProps = {
+    children: messages.pagination[direction],
+    disabled: disabled,
+    [iconPosition]: incrementIcon,
+    minimal: true,
+    size,
+    onClick: handleIncrement.bind(null, incrementForward)
+  };
+  return <Button {...buttonProps} />;
+};
+
+export default class Pages extends PureComponent<Props> {
+  render() {
+    const { hidePageNumbers, ...restProps } = this.props;
+    const IncrementButton = ({ direction }: { direction: string }) => {
```

Comment:
If you pass `restProps` to both uses of IncrementButton below, you can define this outside of `render()`, which is preferable.

Reply:
Yes, thanks for the suggestion! When that is extracted, it actually becomes unnecessary :)